gpytorch/lazy/kronecker_product_added_diag_lazy_tensor.py

```diff
@@ -71,11 +65,11 @@ def _solve(self, rhs, preconditioner=None, num_tridiag=0):
     def _root_decomposition(self):
         evals, q_matrix = self.lazy_tensor.symeig(eigenvectors=True)
         updated_evals = DiagLazyTensor((evals + self.diag_tensor.diag()).pow(0.5))
-        matrix_root = q_matrix.matmul(updated_evals)
+        matrix_root = MatmulLazyTensor(q_matrix, updated_evals)
         return matrix_root
 
     def _root_inv_decomposition(self, initial_vectors=None):
         evals, q_matrix = self.lazy_tensor.symeig(eigenvectors=True)
         inv_sqrt_evals = DiagLazyTensor((evals + self.diag_tensor.diag()).pow(-0.5))
```

Comment:
May want to clamp these at 0 for numerical stability

Reply:
this happens here if I'm not mistaken: https://github.com/cornellius-gp/gpytorch/blob/master/gpytorch/lazy/lazy_tensor.py#L1881